File: test_data/code/python/1403-minimum-subsequence-in-non-increasing-order.py
class Solution:
    def minSubsequence(self, nums: List[int]) -> List[int]:
        nums.sort(reverse=True)
        total_sum = sum(nums)
        result = []
        subsequence_sum = 0
        for num in nums:
            if subsequence_sum + num > total_sum - subsequence_sum:
                result.append(num)
                subsequence_sum += num
            else:
                break
        return result

File: updated_code/code/java/1403-minimum-subsequence-in-non-increasing-order.java
//Package declaration
package test;

import java.util.*;

class Solution {
    //Class comment
    public List<Integer> minSubsequence(int[] nums) {
        //Method comment
        Arrays.sort(nums);
        int sum = 0, left = 0;
        for (int right = nums.length - 1; left < right; ) {
            sum += nums[right--];
            left++;
            while (left < nums.length && sum >= nums[left]) {
                left++;
            }
        }
        List<Integer> res = new ArrayList<>();
        int j = nums.length - 1;
        for (int i = nums.length - 1; i >= left; i--) {
            if (nums[i] > nums[j]) {
                res.add(nums[i]);
                j--;
            }
        }
        return res;
    }
}